localization/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp

```diff
@@ -219,10 +219,21 @@ NDTScanMatcher::NDTScanMatcher()
   sensor_points_sub_ = this->create_subscription<sensor_msgs::msg::PointCloud2>(
     "points_raw", rclcpp::SensorDataQoS().keep_last(points_queue_size),
     std::bind(&NDTScanMatcher::callback_sensor_points, this, std::placeholders::_1), main_sub_opt);
-  regularization_pose_sub_ =
-    this->create_subscription<geometry_msgs::msg::PoseWithCovarianceStamped>(
-      "regularization_pose_with_covariance", 100,
-      std::bind(&NDTScanMatcher::callback_regularization_pose, this, std::placeholders::_1));
+
+  // Only if regularization is enabled, subscribe to the regularization base pose
+  if (regularization_enabled_) {
+    // NOTE: The reason that the regularization subscriber does not belong to the
+    // main_callback_group is to ensure that the regularization callback is called even if
+    // sensor_callback takes time to process.
+    // Also, initial_pose_callback_group is separated from main_callback_group because
+    // callback_initial_pose missed receiving data in the past. resulting in improper interpolation.
+    regularization_pose_sub_ =
+      this->create_subscription<geometry_msgs::msg::PoseWithCovarianceStamped>(
+        "regularization_pose_with_covariance", 10,
+        std::bind(
+          &NDTScanMatcher::callback_regularization_pose, this,
+          std::placeholders::_1) /* Neither main_callback_group nor initial_pose_callback_group */);
```

Comment:
Any reason for not explicitly specifying the third callback_group for this callback?

Reply:
There is no particular reason for the third callback_group. I think we can include it in the `initial_pose_callback_group`.

Reply:
@YamatoAndo I can include it in the `initial_callback_group` rather than`main_callback_group`, right?

Reply:
yes! Please include `regularization_pose_sub_` in the `callback_initial_pose`.

Reply:
@YamatoAndo @kminoda Thank you for your comments! I fixed it in  6c2e70f